perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -2677,6 +2665,20 @@ public List<ApplicationFormItemWithPrefilledValue> getFormItemsWithPrefilledValu
 
 			}
 
+			// prefill USERNAME items with reserved logins
+			if (itemW.getFormItem().getType() == USERNAME && (itemW.getPrefilledValue() == null || itemW.getPrefilledValue().isEmpty())) {
```

Comment:
You should iterate over form items and if this login is prefilled, you should remove password form item for this namespace if its present. We do the same when login is prefilled from attribute or federation.